test/tools/ngsiUtils.js

```diff
@@ -127,17 +144,25 @@ function discoverContextAvailability(host, port, service, subservice, id, type,
     request(options, callback);
 }
 
-function createClient(host, port, service, subservice) {
+function createClient(host, port, ngsiVersion, service, subservice) {
     /*jshint validthis:true */
 
-    return {
-        query: queryEntity.bind(null, host, port, service, subservice),
-        update: updateEntity.bind(null, host, port, service, subservice),
-        discover: discoverContextAvailability.bind(null, host, port, service, subservice)
-    };
+    if (ngsiVersion === 'ld')
+        return {
+            query: queryEntityNgsiLD.bind(null, host, port, service, subservice),
+            update: updateEntity.bind(null, host, port, service, subservice),
+            discover: discoverContextAvailability.bind(null, host, port, service, subservice)
```

Comment:
Those two queries won't work with a NGSI-LD context broker. 

They are not used by the agent?

Reply:
They are not used by the agent indeed. They are test utils but only the query one is used (and removed when integrating nock...)

Reply:
So maybe remove them in the "nock branch" then rebase this branch on top of it.

Reply:
I did it the oppositve way, ngsi-ld-nock is rebased on top of this branch.